test/dom/Value.ts

```diff
@@ -1,161 +1,179 @@
  *                              calling `wrapperConstructor` if provided.
- * @param annotations           An array of strings to associate with the newly constructed value.
  * @param expectedIonType       The new dom.Value's expected Ion type.
- * @param wrapperConstructor    A wrapper constructor for (e.g.) boxing primitives.
+ * @param wrappedValue          A boxed version of jsValue.
  * @param equalityTest          A function that tests whether the new DOM value is equal to the provided jsValue
  */
-function domValueTest(jsValue, annotations, expectedIonType, wrapperConstructor?, equalityTest?) {
-    return () => {
-        let sourceValue = jsValue;
-        if (_hasValue(wrapperConstructor)) {
-            sourceValue = wrapperConstructor(jsValue);
-        }
-        let v: any = Value.from(sourceValue, annotations);
-        let expectedDomType: Constructor = JsValueConversion._domConstructorFor(expectedIonType);
-        if (_hasValue(equalityTest)) {
-            assert.isTrue(equalityTest(jsValue, v));
-        } else {
-            assert.equal(jsValue, v);
-        }
+function domValueTest(jsValue,
+                      expectedIonType,
+                      equalityTest: (p1, p2) => boolean = jsEqualsIon,
+                      wrappedValue?) {
+    function validateDomValue(v: any, annotations) {
+        // Verify that the new dom.Value is considered equal to the original (unwrapped) JS value.
+        assert.isTrue(equalityTest(jsValue, v));
         assert.equal(expectedIonType, v.getType());
+        let expectedDomType: any = JsValueConversion._domConstructorFor(expectedIonType);
         assert.isTrue(v instanceof expectedDomType);
         assert.deepEqual(annotations, v.getAnnotations());
+    }
+
+    return () => {
+        // Test Value.from() with and without specifying annotations
+        let annotations = ['foo', 'bar', 'baz'];
+        let sourceValue = _hasValue(wrappedValue) ? wrappedValue : jsValue;
+        it('' + sourceValue, () => {
+            validateDomValue(Value.from(sourceValue), []);
+        });
+        it(annotations.join('::') + '::' + sourceValue, () => {
+            validateDomValue(Value.from(sourceValue, annotations), annotations);
+        });
     };
 }
 
 describe('Value', () => {
     describe('from()', () => {
-        it('null',
+        describe('null',
             domValueTest(
                 null,
-                ['foo'],
                 IonTypes.NULL,
-                null,
                 (_, domValue) => domValue.isNull()
             )
         );
-        it('boolean',
-            domValueTest(true, ['foo'], IonTypes.BOOL)
+        describe('boolean',
+            domValueTest(true, IonTypes.BOOL)
         );
-        it('Boolean',
-            domValueTest(true, ['foo'], IonTypes.BOOL, Boolean)
+        describe('Boolean',
+            domValueTest(true, IonTypes.BOOL, jsEqualsIon, new Boolean(true))
```

Comment:
My sense is that this becomes clearer if the last parameter is an optional `expectValue`.  The first parameter then always represents the test value :), and _might_ also represent the expected value (except for tests in which an `expectValue` is provided).  These first two tests become:
```typescript
describe('boolean',
    domValueTest(true, IonTypes.BOOL);
);
describe('Boolean',
    domValueTest(new Boolean(true), IonTypes.BOOL, jsEqualsIon, true);
);
```
I'll be ok _(eventually...)_ if you opt to leave as-is.  :)

Reply:
I like it :thumbsup: